src/code/z_skelanime.c

```diff
@@ -0,0 +1,1748 @@
+        entry->types.type4.index = index;
+    }
+}
+
+void SkelAnime_LoadAnimationType5(GlobalContext* globalCtx, Actor* actor, SkelAnime* skelAnime, f32 arg3) {
+    AnimationEntry* entry = SkelAnime_NextEntry(&globalCtx->animationCtx, ANIMATION_TYPE5);
+    
+    if (entry != NULL) {
+        entry->types.type5.actor = actor;
+        entry->types.type5.skelAnime = skelAnime;
+        entry->types.type5.unk08 = arg3;
+    }
+}
+
+/* The next functions are callbacks to loading animations */
+
+void SkelAnime_LinkAnimetionLoaded(GlobalContext* globalCtx, AnimationEntryType0* entry) {
+    osRecvMesg(&entry->msgQueue, NULL, OS_MESG_BLOCK);
+}
+
+void SkelAnime_AnimationType1Loaded(GlobalContext* globalCtx, AnimationEntryType1* entry) {
+    s32 i;
+    Vec3s* dst;
+    Vec3s* src;
+    AnimationEntryRaw* genericEntry = (AnimationEntryRaw*)entry;
+
+    if ((genericEntry->raw[0] & D_801F5AB4) != 0) {
+        return;
+    }
+
+    for (dst = entry->dst, src = entry->src, i = 0; i < genericEntry->raw[1]; i++) {
+        *dst++ = *src++;
+    }
+}
+
+void SkelAnime_AnimationType2Loaded(GlobalContext* globalCtx, AnimationEntryType2* entry) {
+    if ((entry->unk00 & D_801F5AB4) == 0) {
+        SkelAnime_InterpolateVec3s(entry->limbCount, entry->unk04, entry->unk04, entry->unk08, entry->unk0C);
+    }
+}
+
+void SkelAnime_AnimationType3Loaded(GlobalContext* globalCtx, AnimationEntryType3* entry) {
+    s32 i;
+    Vec3s* dst;
+    Vec3s* src;
+    u8* index;
+    AnimationEntryRaw* rawEntry = (AnimationEntryRaw*)entry;
+
+    if ((rawEntry->raw[0] & D_801F5AB4) == 0) {
+        for (dst = entry->dst, src = entry->src, index = entry->index, i = 0; i < rawEntry->raw[1]; i++, dst++, src++) {
+            if (*(index++)) {
+                *dst = *src;
+            }
+        }
+    }
+}
+
+void SkelAnime_AnimationType4Loaded(GlobalContext* globalCtx, AnimationEntryType4* entry) {
+    s32 i;
+    Vec3s* dst;
+    Vec3s* src;
+    u8* index;
+    AnimationEntryRaw* rawEntry = (AnimationEntryRaw*)entry;
+
+    if ((rawEntry->raw[0] & D_801F5AB4) == 0) {
+        for (dst = entry->dst, src = entry->src, index = entry->index, i = 0; i < rawEntry->raw[1]; i++, dst++, src++) {
+            if (*(index++) < 1U) {
+                *dst = *src;
+            }
+        }
+    }
+}
+
+void SkelAnime_AnimationType5Loaded(GlobalContext* globalCtx, AnimationEntryType5* entry) {
+    s32 pad;
+    Actor* actor = entry->actor;
+    Vec3f pos;
+
+    func_80137748(entry->skelAnime, &pos, actor->shape.rot.y);
+    actor->currPosRot.pos.x += (pos.x * actor->scale.x) * entry->unk08;
+    actor->currPosRot.pos.y += (pos.y * actor->scale.y) * entry->unk08;
+    actor->currPosRot.pos.z += (pos.z * actor->scale.z) * entry->unk08;
+}
+
+void func_80135EE8(GlobalContext* globalCtx, AnimationContext* animationCtx) {
+    AnimationEntry* entry = animationCtx->entries;
+
+    for (; animationCtx->animationCount != 0; entry++, animationCtx->animationCount--) {
```

Comment:
Does it not match when `entry` is initialized in the for loop?

Reply:
Unfortunately not. I had to move it outside to match.

Reply:
Changed it to a while loop instead so it seems more natural.